src/main/scala/com/typesafe/sbt/SbtGit.scala

```diff
@@ -117,7 +117,19 @@ object SbtGit {
     gitDescribedVersion := gitReader.value.withGit(_.describedVersion).map(v => git.gitTagToVersionNumber.value(v).getOrElse(v)),
     gitCurrentTags := gitReader.value.withGit(_.currentTags),
     gitCurrentBranch := Option(gitReader.value.withGit(_.branch)).getOrElse(""),
-    gitUncommittedChanges in ThisBuild := gitReader.value.withGit(_.hasUncommittedChanges)
+    gitUncommittedChanges in ThisBuild := gitReader.value.withGit(_.hasUncommittedChanges),
+    scmInfo := {
+      val remote = """origin[ \t]+(?:git@|https?\:\/\/)([^:\/]+)[:\/](.*)\.git[ \t]+\(fetch\)""".r
+      
+      Process("git remote -v").lines_!.collect {
+        case remote(domain, repo) =>
+          ScmInfo(
+            url(s"https://$domain/$repo"),
+            s"scm:git:https://$domain/$repo.git",
```

Comment:
Sadly it seems like this needs to be specified in this way to generate the pom properly. I couldn't make my Sonatype work otherwise.

Regarding the supported URLs, I do agree, though this change is only meant to offer a sane default. Any special cases can still be set manually.